src/Interfaces/AbstractStar.jl

```diff
@@ -0,0 +1,19 @@
+"""
+    AbstractStar{N}
+
+Abstract supertype for all star set types.
+
+### Notes
+
+A set ``X`` is star-like (also known as generalized star) if it can be represented
+by a center ``x₀ ∈ \\mathbb{R}^n`` and ``m`` vectors ``v₁, …, vₘ`` forming the basis,
+and a predicate ``P : \\mathbb{R}^n → \\{⊤, ⊥\\}`` such that
+
+```math
+    X = \\{x ∈ \\mathbb{R}^n : x = x₀ + \\sum_{i=1}^m α_i v_i,,~~\\textrm{s.t. } P(α) = ⊤ \\}.
```

Comment:
What are `α_i` and `α`?

Reply:
i've extended the statement, thanks for checking